### PR TITLE
KBFS/Git Archive Pass 2

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

File: shared/constants/archive.tsx [hunk 1]
```diff
@@ -242,6 +272,84 @@ export const _useState = Z.createZustand<State>((set, get) => {
...
+    set(s => {
+      s.archiveAllFilesResponseWaiter.state = 'waiting'
+    })
+    const f = async () => {
+      const response = await T.RPCGen.SimpleFSSimpleFSArchiveAllFilesRpcPromise({
+        outputDir,
+        overwriteZip: false,
+        includePublicReadonly: false,
+      })
+      console.log({response})
```
Comment (chrisnojima) [inline shared/constants/archive.tsx:299]: 🔥

File: shared/settings/archive/index.tsx [hunk 2]
```diff
@@ -347,8 +366,8 @@ const styles = Kb.Styles.styleSheetCreate(() => ({
   jobs: {
     flexGrow: 1,
     flexShrink: 1,
-    paddingLeft: Kb.Styles.isMobile ? 4 : 16,
-    paddingRight: Kb.Styles.isMobile ? 4 : 16,
+    // paddingLeft: Kb.Styles.isMobile ? 4 : 16,
```
Comment (chrisnojima) [inline shared/settings/archive/index.tsx:369]: 🔥 
did you try these screens on mobile at all?

**Review by strib — APPROVED** [review]
Golang looks good overall, mod nits.

I'm a bit terrified of the "archive all" endpoints, especially having it archive all public favorites.  That could easily swamp someone's local disk very quickly, and I'm not sure it's what most people would want/expect.  But I guess we can try it and see what happens.

File: go/kbfs/simplefs/archive.go [hunk 3]
```diff
@@ -232,6 +234,114 @@ func (m *archiveManager) getCurrentState(ctx context.Context) (
...
+					m.simpleFS.log.CWarningf(ctx, errDesc)
+					pathsWithIssues[itemPath] = errDesc
+					continue loopManifest
+				}
+			}
+		}
+
+		{ // Check hash
+			h := sha256.New()
+			_, err = io.Copy(h, f)
```
[the remark this answers is not in the document]
Comment (songgao) [reply go/kbfs/simplefs/archive.go:326]: It actually skips that when DirentType is a directory. There’s a `continue loopManifest` in that case. Sorry for the long loop!

File: go/kbfs/simplefs/simplefs.go [hunk 4]
```diff
@@ -162,14 +164,73 @@ type SimpleFS struct {
...
+	}
+
+	if k.indexer != nil {
+		err = k.indexer.Shutdown(ctx)
+		if err != nil {
+			return err
+		}
+		k.indexer, err = search.NewIndexer(k.config)
+		if err != nil {
+			k.log.Warning("Couldn't make indexer: %+v", err)
```
[the remark this answers is not in the document]
Comment (songgao) [reply go/kbfs/simplefs/simplefs.go:224]: The `newSimpleFS` function also doesn’t return an error (nor does it call `log.Fatalf` like other errors), so I did the same thing here. Let me know if you want me to change both!

File: go/kbfs/simplefs/archive.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (strib) [reply go/kbfs/simplefs/archive.go:326]: Ah got it, I messed up the nesting in my head.  That makes more sense.

**Pull request merged**